spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -65,6 +82,7 @@ class SparkBatchQueryScan extends SparkPartitioningAwareScan<PartitionScanTask>
   private final Long endSnapshotId;
   private final Long asOfTimestamp;
   private final String tag;
+  private final SparkReadMetricReporter sparkReadMetricReporter;
```

Comment:
What about using `Supplier<ScanReport> scanReportSupplier` to make this independent from a particular metrics reporter? We can use `metricsReporter::scanReport` closure to construct it.

```
InMemoryMetricsReporter metricsReporter = new InMemoryMetricsReporter();
...
scan.metricsReporter(metricsReporter)
...
return new SparkBatchQueryScan(..., metricsReporter::scanReport);
```